pybindsrc/module.cpp

```diff
@@ -8,27 +8,25 @@
 
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
-#include "submodules.hpp"
 
 namespace py = pybind11;
 
-namespace dunedaq {
-namespace detdataformats {
-namespace python {
+namespace dunedaq::detdataformats::python {
```

Comment:
While we've been using extern register_*, it's not great from code/compilation perspective. I would keep the declaration of the register functions in a separate header, changing the name from submodules.hpp to something more appropriate, like registrators.hpp